### PR TITLE
SKALE-3580 Node API 2.0, GA workflow update

### PR DESCRIPTION
**Pull request by dmytrotkk**
In this PR:

- Completly restructured Node API
- Updated all web tests 
- Split test workflows: core/rotation
- Removed old stuff

### CODE REVIEW

File: scripts/install_python_dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (badrogger) [inline scripts/install_python_dependencies.sh:1]: Bad naming for file that spins up sgx and redis containers.

File: tests/docker_utils_test.py [hunk 2]
```diff
@@ -15,15 +15,14 @@
...
-TEST_ABI_FILEPATH = os.path.join(DIR_PATH, 'test_abi.json')
 
 
 @pytest.fixture
 def mocked_client():
     dutils = DockerUtils(volume_driver='local')
     dutils.get_all_schain_containers = partial(
         dutils.get_all_schain_containers,
-        all=True
+        _all=True
```
Comment (badrogger) [inline tests/docker_utils_test.py]: _var syntax is for private variables. To fix linter better to use all_.

File: web/routes/node_exit.py [hunk 3]
```diff
@@ -21,27 +21,24 @@
...
 from flask import Blueprint
 from tools.custom_thread import CustomThread
-from web.helper import construct_ok_response
+from web.helper import construct_ok_response, get_api_url
 
 logger = logging.getLogger(__name__)
+BLUEPRINT_NAME = 'exit'
 
 
 def construct_node_exit_bp(node):
```
Comment (badrogger) [inline web/routes/node_exit.py]: let's consider to merge it with node blueprint. I think skale exit is confusing.

File: web/routes/health.py [hunk 4]
```diff
@@ -19,26 +19,59 @@
...
+        _all = request.args.get('all') == 'True'
+        name_filter = request.args.get('name_filter') or ''
+        containers_list = docker_utils.get_containers_info(
+            _all=_all,
+            name_filter=name_filter,
+            format=True
+        )
+        return construct_ok_response(containers_list)
+
+    @health_bp.route(get_api_url(BLUEPRINT_NAME, 'schains-checks'), methods=['GET'])
```
Comment (badrogger) [inline web/routes/health.py]: May be we can leave just /health/schains instead of schains-checks ?

**Review by badrogger — CHANGES_REQUESTED** [review]

File: scripts/install_python_dependencies.sh (hunk 1 above)
Comment (dmytrotkk) [reply scripts/install_python_dependencies.sh:1]: addressed

File: tests/docker_utils_test.py (hunk 2 above)
Comment (dmytrotkk) [reply tests/docker_utils_test.py]: but otherwise we're redefining built-in `all`

File: web/routes/node_exit.py (hunk 3 above)
Comment (dmytrotkk) [reply web/routes/node_exit.py]: merged

File: web/routes/health.py (hunk 4 above)
Comment (dmytrotkk) [reply web/routes/health.py]: renamed

**Review by badrogger — APPROVED** [review]

**Pull request merged**